### ISSUE TITLE
openjdk-8 is missing liberation_ttf

### ISSUE THREAD

**Issue by pSub** [NixOS/nixpkgs#6305]
I just ran into the following exception

```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at sun.awt.FontConfiguration.getVersion(FontConfiguration.java:1264)
	at sun.awt.FontConfiguration.readFontConfigFile(FontConfiguration.java:219)
	at sun.awt.FontConfiguration.init(FontConfiguration.java:107)
	at sun.awt.X11FontManager.createFontConfiguration(X11FontManager.java:753)
	at sun.font.SunFontManager$2.run(SunFontManager.java:431)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.font.SunFontManager.<init>(SunFontManager.java:376)
	at sun.awt.X11FontManager.<init>(X11FontManager.java:32)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:408)
	at java.lang.Class.newInstance(Class.java:438)
	at sun.font.FontManagerFactory$1.run(FontManagerFactory.java:83)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.font.FontManagerFactory.getInstance(FontManagerFactory.java:74)
	at sun.font.SunFontManager.getInstance(SunFontManager.java:250)
	at sun.font.FontDesignMetrics.getMetrics(FontDesignMetrics.java:264)
	at sun.swing.SwingUtilities2.getFontMetrics(SwingUtilities2.java:1113)
	at javax.swing.JComponent.getFontMetrics(JComponent.java:1623)
	at javax.swing.plaf.basic.BasicGraphicsUtils.getPreferredButtonSize(BasicGraphicsUtils.java:276)
	at javax.swing.plaf.basic.BasicButtonUI.getPreferredSize(BasicButtonUI.java:376)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1659)
	at java.awt.GridLayout.preferredLayoutSize(GridLayout.java:341)
	at java.awt.Container.preferredSize(Container.java:1794)
	at java.awt.Container.getPreferredSize(Container.java:1778)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1661)
	at java.awt.BorderLayout.preferredLayoutSize(BorderLayout.java:719)
	at java.awt.Container.preferredSize(Container.java:1794)
	at java.awt.Container.getPreferredSize(Container.java:1778)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1661)
	at javax.swing.JRootPane$RootLayout.preferredLayoutSize(JRootPane.java:920)
	at java.awt.Container.preferredSize(Container.java:1794)
	at java.awt.Container.getPreferredSize(Container.java:1778)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1661)
	at java.awt.BorderLayout.preferredLayoutSize(BorderLayout.java:719)
	at java.awt.Container.preferredSize(Container.java:1794)
	at java.awt.Container.getPreferredSize(Container.java:1778)
	at java.awt.Window.pack(Window.java:809)
	at scala.swing.Window.contents_$eq(Window.scala:46)
	at gui.CountButton$$anon$2.<init>(a.scala:9)
	at gui.CountButton$.top(a.scala:7)
	at gui.CountButton$.top(a.scala:6)
	at scala.swing.SimpleSwingApplication.startup(SimpleSwingApplication.scala:26)
	at scala.swing.SwingApplication$$anonfun$main$1.apply$mcV$sp(SwingApplication.scala:7)
	at scala.swing.Swing$$anon$10.run(Swing.scala:32)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:744)
	at java.awt.EventQueue.access$400(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:697)
	at java.awt.EventQueue$3.run(EventQueue.java:691)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:714)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

According to https://bugs.launchpad.net/linaro-oe/+bug/1243879 it looks like the dependency `liberation_ttf` is missing.

**Comment by cocreature** [comment]
I remember having similar bugs and I tried compiling against fontconfig and similar stuff but I wasn't able to get it to work. What finally did the trick for me was putting gtk in `LD_LIBRARY_PATH`.

Does it work for you if you use and then run what ever you are trying to run in nix-shell? I have to admit that I submitted the openjdk8 package so it's probably my fault, but I am new to nix and not at all an openjdk expert, so I'm not sure I can do much. If you could give me an easy way to reproduce this I could maybe take a look.
```
{ pkgs ? import <nixpkgs> {} }:
with pkgs;
stdenv.mkDerivation {
  name = "lalala";

  buildInputs = [ gtk ];

  shellHook = ''
    export LD_LIBRARY_PATH="${gtk}/lib:$LD_LIBRARY_PATH"
  '';
}
```

**Comment by pSub** [comment]
Thanks, I am currently busy and will look into this again by Tuesday at the latest.

**Comment by purefn** [comment]
I'm running into this same exception too. To reproduce just download and try and run Intellij IDEA. It happens everytime I try and start IDEA.  I tried using `nix-shell` with the expression above, but that didn't help.

**Comment by cocreature** [comment]
I am sorry, but I stopped using nix, so I can't help you there. I should probably make a pr to remove myself from the maintainer list. Pinging @shlevy as he seems to maintain openjdk7.

**Comment by edwtjo** [comment]
Fixed in ad03fb11b353186735112385d2ec67bc76a06f07

**Comment by edwtjo** [comment]
@cocreature I've removed you from the maintainer list as you requested. If you have the time and don't mind; why did you drop nix? Thank you for your contribution to nixpkgs.
